import_export/resources.py

```diff
@@ -438,18 +438,24 @@ def validate_instance(self, instance, import_validation_errors=None, validate_un
         if errors:
             raise ValidationError(errors)
 
-    def save_instance(self, instance, using_transactions=True, dry_run=False):
+    def save_instance(self, instance, is_create, using_transactions=True, dry_run=False):
```

Comment:
As far as I can tell, this is a change in the API and would be retro incompatible. Right?

Reply:
Good catch. I don't think we can keep backwards compatibility without complicating the code too much. I'd say we bump the major version when releasing it.

Reply:
Good point.  It doesn't seem right to modify the API to fix this issue.  Also, I need to rethink this one because there are a couple of other places which refer to `instance.pk`.

One idea would be to provide a hook to return a "pk_field" property which defines the pk field to use, so that anyone hitting this issue (i.e. those that use a different PK field) can set this to override the use of `instance.pk`.

I'll give this some thought and try a different approach.

Reply:
What about this?

```python
    def save_instance(self, instance, using_transactions=True, dry_run=False, new=None):
        ...
        if self._meta.use_bulk:
            if new is None:
                raise Exception('save_instance needs the `new` parameter when using bulk mode')
            if new:
                ...
```

API is not modified, only people using `use_bulk` **and** overriding `import_row` will hit the exception, and still they get a descriptive message about the problem. They should be using `before_import` and `after_import`, as `import_row` is quite complex already. I think this is better than the current situation in which `use_bulk` fails silently for some people.

Tried this idea on top of your branch and all tests are ok.

Reply:
Have marked this as 'breaking change', to be scheduled for release 3.0